File: android/lib/src/main/java/duck/cameras/android/model/PtzData.java
package duck.cameras.android.model;

import java.io.Serializable;
import java.util.ArrayList;

public class PtzData implements Serializable {
    public Camera camera;
    public Camera.Profile profile;
    public ArrayList<Configuration> configurations;

    public PtzData() {
        configurations = new ArrayList<>();
    }

    public static class Configuration implements Serializable {
        public String name;
        public String token;
        public RelativeSpace relativeSpace;
        public ContinuousSpace continuousSpace;
        public SpeedSpace speedSpace;

        public Configuration() {
            relativeSpace = new RelativeSpace();
            continuousSpace = new ContinuousSpace();
            speedSpace = new SpeedSpace();
        }
    }

    public static class RelativeSpace implements Serializable {
        public Float xmin;
        public Float xmax;
        public Float ymin;
        public Float ymax;
        public Float zmin;
        public Float zmax;
    }

    public static class ContinuousSpace implements Serializable {
        public Float xmin;
        public Float xmax;
        public Float ymin;
        public Float ymax;
        public Float zmin;
        public Float zmax;
    }

    public static class SpeedSpace implements Serializable {
        public Float xmin;
        public Float xmax;
        public Float ymin;
        public Float ymax;
        public Float zmin;
        public Float zmax;
    }

}

File: windows/Model/PtzData.cs
using System.Collections.Generic;

namespace Duck.Cameras.Windows.Model
{
    public class PtzData
    {
        public Camera Camera;
        public CameraProfile Profile;
        public List<PtzConfiguration> Configurations;

        public PtzData()
        {
            Configurations = new List<PtzConfiguration>();
        }
    }

    public class PtzConfiguration
    {
        public string Name;
        public string Token;
        public PtzRelativeSpace RelativeSpace;
        public PtzContinuousSpace ContinuousSpace;
        public PtzSpeedSpace SpeedSpace;

        public PtzConfiguration()
        {
            RelativeSpace = new PtzRelativeSpace();
            ContinuousSpace = new PtzContinuousSpace();
            SpeedSpace = new PtzSpeedSpace();
        }
    }

    public class PtzRelativeSpace
    {
        public float? XMin;
        public float? XMax;
        public float? YMin;
        public float? YMax;
        public float? ZMin;
        public float? ZMax;
    }

    public class PtzContinuousSpace
    {
        public float? XMin;
        public float? XMax;
        public float? YMin;
        public float? YMax;
        public float? ZMin;
        public float? ZMax;
    }

    public class PtzSpeedSpace
    {
        public float? XMin;
        public float? XMax;
        public float? YMin;
        public float? YMax;
        public float? ZMin;
        public float? ZMax;
    }

}
